flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -113,6 +114,9 @@ public FlinkCatalog(
     asNamespaceCatalog =
         originalCatalog instanceof SupportsNamespaces ? (SupportsNamespaces) originalCatalog : null;
     closeable = originalCatalog instanceof Closeable ? (Closeable) originalCatalog : null;
+
+    EnvironmentContext.put(EnvironmentContext.ENGINE_NAME, "flink");
+    EnvironmentContext.put(EnvironmentContext.ENGINE_VERSION, "1.14");
```

Comment:
would be great if we'd have a way to programmatically fetch the version here and in other places

Reply:
Can't we leverage the package information?

Reply:
Could you please elaborate on that? Do you mean reading the version from the file structure? The jar itself doesn't have that file structure as we have in the project (there's no 1.x in the file structure).

Reply:
we probably need to figure out the programmatical way. otherwise, next time when we add a new module like 1.17. This hard-coded value can easily be missed and carried over.